### PR TITLE
[CARBONDATA-3027] Increase unsafe working memory default size and add log file for SDK

### PR DESCRIPTION
**Pull request by xubo245**
When I read 1.42G data from local  by SDK carbon reader , it throw exception:

```
java.lang.RuntimeException: org.apache.carbondata.core.memory.MemoryException: Not enough memory. please increase carbon.unsafe.working.memory.in.mb
	at org.apache.carbondata.core.datastore.page.UnsafeDecimalColumnPage.putBytes(UnsafeDecimalColumnPage.java:167)
	at org.apache.carbondata.core.datastore.page.VarLengthColumnPageBase.getVarLengthColumnPage(VarLengthColumnPageBase.java:281)
	at org.apache.carbondata.core.datastore.page.VarLengthColumnPageBase.getComplexLVBytesColumnPage(VarLengthColumnPageBase.java:250)
	at org.apache.carbondata.core.datastore.page.VarLengthColumnPageBase.newComplexLVBytesColumnPage(VarLengthColumnPageBase.java:157)
	at org.apache.carbondata.core.datastore.page.ColumnPage.newComplexLVBytesPage(ColumnPage.java:335)
```
other users also has this problem. 512 is not enough for them. They change to 1024, it's working. I also test in local, it's the same. 
I suggest increase carbon.unsafe.working.memory.in.mb default  value 512M to 1024M, which can avoid more related problem.

what's more, when we test in local, there is not log for analysisinp detail of this error. So I add log for it.

This PR
1.Increase unsafe working memory default size from 512m to 1024m
2.add log file for SDK

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [ ] Document update required?
Yes
 - [x] Testing done
     No
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/877/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9142/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1075/

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 1]
```diff
@@ -1234,7 +1234,7 @@
 
   @CarbonProperty
   public static final String UNSAFE_WORKING_MEMORY_IN_MB = "carbon.unsafe.working.memory.in.mb";
-  public static final String UNSAFE_WORKING_MEMORY_IN_MB_DEFAULT = "512";
+  public static final String UNSAFE_WORKING_MEMORY_IN_MB_DEFAULT = "2048";
```
Comment (xuchuanyin) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: What will happen if user do not have that much memory?

Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: I don't know.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9143/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1076/

**Comment by xuchuanyin** [comment]
so we should not change this value without better knowledge about it.
If you really want to change it, the corresponding documents are needed to be changed at the same time.

If you just counter some failures in your test, you can set/unset this property value to avoid modifying the default value.

**Comment by jackylk** [comment]
Can you describe a issue you encounter using the original default value?

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 2]
```diff
@@ -1234,7 +1234,7 @@
 
   @CarbonProperty
   public static final String UNSAFE_WORKING_MEMORY_IN_MB = "carbon.unsafe.working.memory.in.mb";
-  public static final String UNSAFE_WORKING_MEMORY_IN_MB_DEFAULT = "512";
+  public static final String UNSAFE_WORKING_MEMORY_IN_MB_DEFAULT = "1024";
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: Can you describe the issue you encountered using the original default value?

File: store/sdk/src/main/resources/log4j.properties [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Root logger option
+log4j.rootLogger=INFO,stdout
+
+
+# Redirect log messages to console
+log4j.appender.debug=org.apache.log4j.RollingFileAppender
+log4j.appender.stdout=org.apache.log4j.ConsoleAppender
+log4j.appender.stdout.Target=System.out
+log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+log4j.appender.stdout.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n
```
Comment (jackylk) [inline store/sdk/src/main/resources/log4j.properties]: Now we are using log4j Logger directly, you can add %C also in the pattern to print the class name

**Comment by xubo245** [comment]
@xuchuanyin @jackylk I added some describe in ahead about this issue.

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 2 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: ok, I described the issue in ahead

File: store/sdk/src/main/resources/log4j.properties (hunk 3 above)
Comment (xubo245) [reply store/sdk/src/main/resources/log4j.properties]: there is already add  %c{1} to print class name:
```
2018-10-30 14:26:23 INFO  CacheProvider:168 - Executor LRU cache size not configured. Initializing with driver LRU cache size.
2018-10-30 14:26:23 INFO  CarbonLRUCache:76 - LRU cache size not configured. Therefore default behavior will be considered and no LRU based eviction of columns will be done
2018-10-30 14:26:23 INFO  SegmentPropertiesAndSchemaHolder:117 - Constructing new SegmentProperties for table: null_null. Current size of segment properties holder list is: 1
2018-10-30 14:26:23 INFO  QueryModelBuilder:94 - Projection Columns: [stringfield, datefield, timefield, varcharfield, arrayfield, shortfield, intfield, longfield, doublefield, boolfield, decimalfield, floatfield]
2018-10-30 14:26:23 INFO  AbstractQueryExecutor:121 - Query will be executed on table: null
2018-10-30 14:26:23 INFO  ResultCollectorFactory:79 - Row based dictionary collector is used to scan and collect the data
2018-10-30 14:26:24 WARN  DataMapStoreManager:499 - failed to get carbon table from table PathFile does not exist: /Users/xubo/Desktop/xubo/git/carbondata1/store/sdk/carbondata/Metadata/schema
2018-10-30 14:26:24 INFO  CarbonLRUCache:163 - Removed entry from InMemory lru cache :: /Users/xubo/Desktop/xubo/git/carbondata1/store/sdk/carbondata/632392937014916_batchno0-0-null-632392085265031.carbonindex
2018-10-30 14:26:24 INFO  UnsafeMemoryManager:175 - Total offheap working memory used after task 14e4fd01-9df5-416a-9b0a-6970ea63c2bd is 0. Current running tasks are d26d0593-f4ce-4bb8-ac94-974638229901, a92307f4-9292-4645-87fb-2a772b421df4

```

**Comment by xubo245** [comment]
@xuchuanyin updated doc

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1142/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1354/

**Comment by KanakaKumar** [comment]
I think this PR may not be require any more after #2841.  @xubo245 , our intention is to make unsafe configuration flexible and even with default values we should  be able to continue with heap memory.

**Comment by xubo245** [comment]
OK，I tested， it's work。 I  will close this PR.

**Pull request closed without merging**